### ISSUE TITLE
Regression: order(nil) no longer works

### ISSUE THREAD

**Issue by levent** [toptal/chewy#356]
Regression in 0.8.4

In 0.8.3 order(nil) would silently work. Now it tries to pass that onto ES resulting in a valid exception.
I'm not sure if this is/was expected behaviour but thought I'd raise it just in case. I think it's common for users to do something like...

```ruby
UsersIndex.query(term: {name: 'foo'}).order(params[:order]) # sanitized params ideally where it could be left blank and result in nil.
```

```bash
[400] {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"malformed sort format, within the sort array, an object, or an actual string are allowed"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"test_app","node":"_apI_g7gQ4WT-vbWdiqyvQ","reason":{"type":"illegal_argument_exception","reason":"malformed sort format, within the sort array, an object, or an actual string are allowed"}}]},"status":400}
```

**Comment by pyromaniac** [comment]
I don't remember I did something with DSL between releases, did you check elasticsearch-ruby for changed behavior?

**Comment by levent** [comment]
No other gems or ES itself have changed.
I'll try and investigate further.

**Comment by pyromaniac** [comment]
Would be perfect, thanks!

**Comment by levent** [comment]
This is a very crude experiment but I forked both tags (v0.8.4 and v0.8.3) and changed one line in the kaminari spec: https://github.com/toptal/chewy/commit/0aebb6f3a2970f5922cdb2f6f6ab4a046006aeeb

Tests in v0.8.3 - https://travis-ci.org/levent/chewy/jobs/118462121
Tests in v0.8.4 - https://travis-ci.org/levent/chewy/jobs/118461886

**Comment by pyromaniac** [comment]
Could I ask you to submit a fix please? I frankly have no idea when will I be able to fix this, definitely not earlier the next weekend. Yeah, regression seems to be exists somehow :( Have no idea how though.

**Comment by levent** [comment]
I'll try and find some time this week :+1:

**Comment by pyromaniac** [comment]
Would be great, thanks!

**Comment by levent** [comment]
Apologies for the delay!

I finally found the issue. I wonder what you propose?

See https://github.com/toptal/chewy/blob/master/lib/chewy/query/criteria.rb#L117
```sort?``` returns ```true``` when the value of ```sort``` is ```[nil]```

This is happening because ```sort?``` is using ```sort.present?``` now instead of ```sort.any?``` (This applies to all the STORAGE methods. You may want to compact all the arrays first but this may go against the performance optimisations you made in the commit that introduced the bug.

The relevant commit introducing the regression is https://github.com/toptal/chewy/commit/0d25b1d889d76e1fb55f5f97ac0da27f0e192414

**Comment by pyromaniac** [comment]
Wow, such an investigation, thank you so much! Could I ask you to change it back for sorting and maybe add a spec to secure this behavior in the future?

**Comment by daniloisr** [comment]
This issue is fixed by #388, we can close it now :]

**Comment by pyromaniac** [comment]
Thanks!
